### DISCUSSION TITLE
Asset Tag - Show Folder with subdirectories

### DISCUSSION THREAD

**Discussion by nicorttngr** [statamic/cms#6035] [Q&A]
Hi!

I'm trying to display assets from my private S3 Bucket in frontend which is working fine but I also would like to show the folders in the current directory. But as it seems the Assets Tag only retrieves the assets itself. My goal is to show a library-like structure of the assets where the user can click through the folders. Is this possible with the core functions?

**Comment by edalzell** [comment]
I made a custom tag for that for a project

**Reply by nicorttngr** [reply]
You mean a tag that returns the directory structure by working with Laravels native Filesystem functions - so no need for the asset loop?

**Reply by edalzell** [reply]
> You mean a tag that returns the directory structure by working with Laravels native Filesystem functions - so no need for the asset loop?

ya
